### PR TITLE
Advisory: rust-openssl hostname verification

### PR DESCRIPTION
**Pull request by tarcieri**
cc @sfackler

### CODE REVIEW

**Comment by tarcieri** [comment]
@sfackler do the contents of this advisory look correct to you?

**Comment by sfackler** [comment]
The version number seems wrong - 1.0.2 was when hostname verification support was added to OpenSSL proper. The rust-openssl 0.9.0 release had both the bindings in that PR as well as a handrolled implementation when linking against 1.0.1 which were set up when using the `SslConnector` type which was added in that release as well.

**Comment by sfackler** [comment]
The separate [openssl-verify](https://crates.io/crates/openssl-verify) crate provided the handrolled implementation in a way compatible with rust-openssl 0.7 and 0.8, which Hyper used before its 0.10 release.

**Comment by tarcieri** [comment]
@sfackler should it be this then?

```toml
patched_versions = [">= 0.9.0"]
```

**Comment by sfackler** [comment]
I think it might make sense to make the advisory more generally about the lack of good defaults before 0.9 - hostname verification is part of that, but so is enabling certificate verification in the first place, limiting cipher suites, setting up forward secrecy on the server side, etc.

**Comment by tarcieri** [comment]
@sfackler can you take another look at this? I tried to update it per your suggestions

File: crates/openssl/RUSTSEC-0000-0000.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+description = """
+All versions of rust-openssl prior to 0.9.0 contained numerous insecure defaults
+including off-by-default certificate verification and no API to perform hostname
+verification.
+
+Unless configured correctly by a developer, these defaults could allow an attacker
+to perform man-in-the-middle attacks.
+
+The problem was addressed in newer versions by enabling certificate verification
+by default and exposing APIs to perform hostname verification.
```
Comment (sfackler) [inline crates/openssl/RUSTSEC-0000-0000.toml]: It's probably worth noting here that the good defaults are present when you use the `SslConnector` and `SslAcceptor` types, not the more low-level `SslContext` type.

Comment (tarcieri) [reply crates/openssl/RUSTSEC-0000-0000.toml]: Added, thanks!

**Pull request merged**